### ISSUE TITLE
Support for Chrome desktop mode on iOS

### ISSUE THREAD

**Issue by Andekas** [shaka-project/shaka-player#5141]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**

Yes

**What version of Shaka Player are you using?**

Latest

**Can you reproduce the issue with our latest release version?**

Yes

**Can you reproduce the issue with the latest code from `main`?**

Yes

**Are you using the demo app or your own custom app?**

Both

**If custom app, can you reproduce the issue using our demo app?**

**What browser and OS are you using?**

iOS (iPad, iPhone), Chrome

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**

**What are the manifest and license server URIs?**

Happens on demo page also

**What configuration are you using?  What is the output of `player.getConfiguration()`?**

Happen on demo page also

**What did you do?**

Open demo bage on iOS Chrome, request desktop site. 

**What did you expect to happen?**

Videos should play

**What actually happened?**

Browser is not supported

**Comment by theodab** [comment]
See the notes on our support matrix:

> We support iOS 13+ through Apple's native HLS player. We provide the same top-level API, but we just set the video's src element to the manifest/media. So we are dependent on the browser supporting the manifests.

Shaka Player will not be able to play every demo asset on iOS, as we lack access to the browser APIs that allow the player to truly run on that platform. Instead, only things that iOS can handle natively are supported (e.g. HLS).

**Comment by Andekas** [comment]
Yes, I tested with HLS stream. Currently, the video plays on iOS, if mobile page is requested. But when using "Request desktop site" video won't play.

**Comment by theodab** [comment]
Oh, I see, I had misread your issue.
I hadn't heard of this "request desktop mode" feature before, but I tried it out and you're right that it makes the browser fail.

Anyway, testing this on my iPhone... according to [the support diagnostic page](https://shaka-player-demo.appspot.com/support.html), the user agent goes from being:
```
Mozilla/5.0 (iPhone; CPU iPhone OS 16_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/111.0.5563.101 Mobile/15E148 Safari/604.1
```
To being:
```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/111 Version/11.1.1 Safari/605.1.15
```
I don't have my stuff for debugging live iPhones on me at the moment, but looking at that spoofed user agent, I think we would identify that as being Safari version 11, due to the `Version/11.1.1` within the string. We explicitly don't support Safari versions before 13, so we reject it.

I double-checked that userAgent string on an online user agent parsing website, and it identified it as being "Chrome 111 on macOS (High Sierra)", because of the `CriOS/111` within the string.

So basically, it looks like the "request desktop mode" feature is tricking us into thinking that the browser is an old version of Safari running on a desktop computer. Instead of tricking us into thinking that the browser is a new version of Chrome running on a desktop computer, like the feature is supposed to.
So we probably need to update our user agent checking code to account for this feature.

**Comment by avelad** [comment]
It should be fixed in v.4.6. Thanks!
